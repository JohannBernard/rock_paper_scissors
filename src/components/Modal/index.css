.modal_window {
    position: fixed;
    background-color: rgba(0, 0, 0, 0.25);
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    z-index: 999;
    visibility: hidden;
    opacity: 0;
    pointer-events: none;
    transition: all 0.3s;
}
.modal_window_visible {
    visibility: visible;
    opacity: 1;
    pointer-events: auto;
}
.modal_container{
    width: 400px;
    max-width: 75vw;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    padding: 32px;
    background: white;
    border-radius: 1rem;
    display: flex;
    flex-direction: column;
    align-items: center;
    box-sizing: border-box;
}
.modal_label {
    font-size: 32px;
    margin-bottom: 32px;
    font-family: 'Poppins', sans-serif;
}

/* Responsive */
@media screen and (max-width: 800px) {
    .modal_label {
        font-size: 28px;
    }
}