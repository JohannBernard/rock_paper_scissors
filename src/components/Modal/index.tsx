import './index.css'
import Button from '../Button'

interface IModal {
    label?: string
    open: boolean
    onClickCustom?: React.MouseEventHandler<HTMLButtonElement>
    id?: number
}

function Modal(props: IModal) {
    return (
        <div
            id="open-modal"
            className={
                'modal_window' + (props.open ? ' modal_window_visible' : '')
            }
            data-testid={props.id ? 'testModal-' + props.id : ''}
        >
            <div className="modal_container">
                <div
                    className="modal_label"
                    data-testid={props.id ? 'testModalLabel-' + props.id : ''}
                >
                    {props.label}
                </div>
                <Button
                    label="Recommencer"
                    onClickCustom={props.onClickCustom}
                />
            </div>
        </div>
    )
}

export default Modal
