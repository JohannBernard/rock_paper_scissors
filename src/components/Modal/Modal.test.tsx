import { render, screen, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import Modal from './index'

afterEach(() => {
    cleanup()
})

test('should render modal component visible', () => {
    render(<Modal label={'Test Modal 1'} open={true} id={1} />)
    const modalElement = screen.getByTestId('testModal-1')
    expect(modalElement).toBeInTheDocument()
    expect(modalElement).toHaveClass('modal_window_visible')
    const modalLabelElement = screen.getByTestId('testModalLabel-1')
    expect(modalLabelElement).toBeInTheDocument()
    expect(modalLabelElement).toHaveTextContent('Test Modal 1')
})

test('should render modal component hidden', () => {
    render(<Modal label={'Test Modal 2'} open={false} id={2} />)
    const itemIconElement = screen.getByTestId('testModal-2')
    expect(itemIconElement).toBeInTheDocument()
    expect(itemIconElement).not.toHaveClass('modal_window_visible')
})

test('matches item snapshot', () => {
    const tree = renderer
        .create(
            <>
                <Modal label={'Test Modal 1'} open={true} id={1} />
                <Modal label={'Test Modal 2'} open={false} id={2} />
            </>
        )
        .toJSON()
    expect(tree).toMatchSnapshot()
})
