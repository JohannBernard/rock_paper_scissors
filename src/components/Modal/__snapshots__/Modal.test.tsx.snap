// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches item snapshot 1`] = `
Array [
  <div
    className="modal_window modal_window_visible"
    data-testid="testModal-1"
    id="open-modal"
  >
    <div
      className="modal_container"
    >
      <div
        className="modal_label"
        data-testid="testModalLabel-1"
      >
        Test Modal 1
      </div>
      <button
        className="my_coach_button"
        data-testid="testButton"
      >
        Recommencer
      </button>
    </div>
  </div>,
  <div
    className="modal_window"
    data-testid="testModal-2"
    id="open-modal"
  >
    <div
      className="modal_container"
    >
      <div
        className="modal_label"
        data-testid="testModalLabel-2"
      >
        Test Modal 2
      </div>
      <button
        className="my_coach_button"
        data-testid="testButton"
      >
        Recommencer
      </button>
    </div>
  </div>,
]
`;
